### ISSUE TITLE
Add a way to build a static library with corrosion_add_cxxbridge

### ISSUE THREAD

**Issue by noufelez** [corrosion-rs/corrosion#363]
I'm currently using CXX and corrosion to interop between c++ and rust. I build rust code as a library and link it to my other c++ libraries and executable.

The target generated by corrosion_add_cxxbridge is a shared library (.so), I was wondering if there would be an option to generate a static library (.a)  ? 

Thanks! :)

**Comment by jschwe** [comment]
> The target generated by corrosion_add_cxxbridge is a shared library (.so)

This depends on what [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS) is set to. Behind the scenes corrosion currently just does an 
[`add_library(${cxx_target})`](https://github.com/corrosion-rs/corrosion/blob/0032ad4b9629737d5a6790b0ecd9d8941a1280bf/cmake/Corrosion.cmake#LL1517C2-L1517C2).

I think it would make sense to add an optional parameter to `corrosion_add_cxxbridge` to pass to the `add_library`  call.

Please also be aware that `corrosion_add_cxxbridge` is still somewhat experimental (it's not in an officially released version yet, only on master) and still needs some polishing and potentially bugfixing before it's ready for a release. PRs and testing is defintely very welcome though!

About the warning message - I recommend reading the linking section of the documentation: https://corrosion-rs.github.io/corrosion/advanced.html#linking
Basically `corrosion_link_libraries` has no effect when called on a Rust static-lib. If your final artifact (usually an executable)  is a C++ artifact, you should just use `target_link_libraries()` instead. If your final artifact is a Rust binary, you should call corrosion_link_libraries with the target name of the Rust binary, and not the Rust static lib.

**Comment by noufelez** [comment]
Thanks for your quick response! It makes a lot more sense now.  I'll continue using corrosion and reading the source code to get more familiar and hopefully start contributing! :)
